rancher2/structure_node_template.go

```diff
@@ -2,10 +2,27 @@ package rancher2
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 )
 
+func fixNodeTempateID(current, id string) string {
+	if len(current) == 0 || len(id) == 0 {
+		return ""
+	}
+	// Fix breaking API change https://github.com/rancher/rancher/pull/23718
+	limitVersion := "2.3.3"
+	if ok, _ := IsVersionGreaterThanOrEqual(current, limitVersion); ok {
+		prefix := "cattle-global-nt:nt-"
+		if !strings.HasPrefix(id, prefix) {
```

Comment:
I would maybe just say if it doesn't have prefix "cattle-global-nt:" to future proof. who knows if we'll ever make a nodetemplate without "nt-" prefix.

Reply:
What do you propose??